include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf.h

```diff
@@ -80,6 +80,53 @@ class TimeIntBDF : public TimeIntBDFBase<Number>
   get_velocities_and_times_np(std::vector<VectorType const *> & velocities,
                               std::vector<double> &             times) const;
 
+  void
+  get_pressures_and_times(std::vector<VectorType const *> & pressures,
+                          std::vector<double> &             times) const;
+
+  void
+  get_pressures_and_times_np(std::vector<VectorType const *> & pressures,
+                             std::vector<double> &             times) const;
+
+  /**
+   * if prepare_vectors_for_next_timestep() was called before
+   * get_quantities_and_times_np() obtained vectors are not sorted
+   * according to times
+   * this function sorts quantites as //[q(t_np), q(t_n), ...]
+   */
+  void
+  sort_quantities_according_to_times(std::vector<VectorType const *> & quantities,
+                                     std::vector<double> const &       times)
+  {
+    // find sort permutation of times
+    std::vector<unsigned int> permutation(times.size());
+    std::iota(permutation.begin(), permutation.end(), 0);
+    std::sort(permutation.begin(), permutation.end(), [&](auto const i, auto const j) {
+      return times[i] > times[j];
+    });
+
+    // apply permutation to vector of quantities
+    std::vector<VectorType const *> unsorted_quantities = quantities;
+
+    std::transform(permutation.begin(), permutation.end(), quantities.begin(), [&](auto const i) {
+      return unsorted_quantities[i];
+    });
+  }
```

Comment:
This should not be public functionality of the time integrator. It should rather appear in the `Driver` class if this is needed (even though I still do not know the concrete reason for this ...).

The function `prepare_vectors_for_next_timestep()` is done in `do_timestep_post_solve()` or `advance_one_timestep_post_solve()`, right? Hence, one should in the `Driver` extract the pressure time derivative at the point of the algorithm where it is available. 

I don't like implementations where we kind of "post-correct" things because we do not want to think about a proper design.